src/hipscat/catalog/catalog.py

```diff
@@ -1,43 +1,71 @@
-        self.catalog_path = catalog_path
-        self.catalog_base_dir = file_io.get_file_pointer_from_path(catalog_path)
-        self.metadata_keywords = None
+class Catalog(Dataset):
+    """A HiPSCat Catalog with data stored in a HEALPix Hive partitioned structure
 
-        self.partition_info = None
-        self.catalog_info = None
+    Catalogs of this type are partitioned spatially, contain `partition_info` metadata specifying
+    the pixels in Catalog, and on disk conform to the parquet partitioning structure
+    `Norder=/Dir=/Npix=.parquet`
+    """
 
-        self.catalog_name = None
-        self.catalog_type = None
+    CatalogInfoClass = CatalogInfo
+    PixelInputTypes = Union[pd.DataFrame, PartitionInfo]
+    HIPS_CATALOG_TYPES = [CatalogType.OBJECT, CatalogType.SOURCE, CatalogType.MARGIN]
 
-        self._initialize_metadata()
+    def __init__(
+        self,
+        catalog_info: CatalogInfoClass,
+        pixels: PixelInputTypes,
+        catalog_path=None,
+    ) -> None:
+        """Initializes a Catalog
 
-    def _initialize_metadata(self):
-        if not file_io.does_file_or_directory_exist(self.catalog_base_dir):
-            raise FileNotFoundError(
-                f"No directory exists at {str(self.catalog_base_dir)}"
-            )
-        catalog_info_file = paths.get_catalog_info_pointer(self.catalog_base_dir)
-        if not file_io.does_file_or_directory_exist(catalog_info_file):
-            raise FileNotFoundError(
-                f"No catalog info found where expected: {str(catalog_info_file)}"
+        Args:
+            catalog_info: CatalogInfo object with catalog metadata
+            pixels: Specifies the pixels contained in the catalog. Can be either a Dataframe with
+                columns `Norder`, `Dir`, and `Npix` matching a `partition_info.csv` file, or a
+                PartitionInfo object
+            catalog_path: If the catalog is stored on disk, specify the location of the catalog
+                Does not load the catalog from this path, only store as metadata
+        """
+        if catalog_info.catalog_type not in self.HIPS_CATALOG_TYPES:
+            raise ValueError(
+                f"Catalog info `catalog_type` must be one of "
+                f"{', '.join([t.value for t in self.HIPS_CATALOG_TYPES])}"
             )
-        self.catalog_info = read_from_metadata_file(catalog_info_file)
-        self.catalog_name = self.catalog_info.catalog_name
-        self.catalog_type = self.catalog_info.catalog_type
+        super().__init__(catalog_info, catalog_path)
+        self.partition_info = self._get_partition_info_from_pixels(pixels)
+        self.pixel_tree = self._get_pixel_tree_from_pixels(pixels)
 
-        if self.catalog_type in ("object", "source"):
-            self.partition_info = PartitionInfo(self.catalog_base_dir)
+    @staticmethod
+    def _get_partition_info_from_pixels(pixels: PixelInputTypes) -> PartitionInfo:
+        if isinstance(pixels, PartitionInfo):
```

Comment:
Should we just pick which is the preferred input type and go with it? Are there compelling reasons to accept both types?